open_instruct/dataset_transformation.py

```diff
@@ -1180,9 +1317,44 @@ def __post_init__(self):
 
     def update_range(self, dataset_range: int):
         self.dataset_range = dataset_range
-        if self.dataset_range > len(self.dataset):
-            raise ValueError("Dataset range exceeds dataset length")
-        self.dataset = self.dataset.select(range(self.dataset_range))
+        original_size = len(self.dataset)
+        self.original_dataset_size = original_size
+        
+        if self.dataset_range <= original_size:
+            # Normal case: select subset
+            self.dataset = self.dataset.select(range(self.dataset_range))
```

Comment:
nit-ish: can we shuffle the dataset or indices a bit when we sub-select? I think random subselection is a better default than taking first n.

Reply:
yup, good call. this actually got me to simplify the logic, so now we just always call the same select_samples function, and it'll handle the <1.0 case as well as the >=1.0 case